chainstate/test-framework/src/utils.rs

```diff
@@ -304,3 +306,78 @@ pub fn pos_mine(
     }
     None
 }
+
+pub fn sign_witnesses(
+    key_manager: &KeyManager,
+    chainstate: &TestChainstate,
```

Comment:
It's confusing that data is retrieved from both chainstate and InMemory* stores. I think you can remove chainstate and get all data for PoS and tokens from InMemory*. (passing &ChainConfig directly)

Reply:
we need both because InMemory Store is updated while the Tx is being created, so if a token authority is changed it will change in the store, but if a previous Tx in the same block uses the same token it needs the previous authority to sign the input.
Which now I realize if 2 tx change the authority in the same block this will fail...

Reply:
pos store is used for delegations, for pools if the decommission happens from a CreateStakPool then I take the key from the utxo itself, and if it is a ProduceFromBlock I take it from chainstate as it means it comes from a previous block.
Also my bad for thinking the token can be wrong as the signing happens per Transaction so the store will have the right info at that time of singing.

Reply:
right. Let's add some comments there to avoid confusion in the future